practice-two/src/pages/Analytics/Analytics.jsx

```diff
@@ -1,34 +1,46 @@
 // Library
-import React, { useState } from "react";
-import { v4 as uuidv4 } from "uuid";
-import useSWR from "swr";
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import useSWR from 'swr';
 
 // Import style for Analytics component
-import "./Analytics.css";
+import './Analytics.css';
 
 // Import service to call API
-import { getAllCustomerService } from "@services";
+import { getAllCustomerService } from '@services';
 
 // Import images or icons
-import { plusIcon, loadingData } from "@assets/images";
+import { plusIcon, loadingData } from '@assets/images';
 
 // Import components
-import { Button, SortData, CustomerItem, Toast } from "@components";
+import {
+  Button,
+  SortData,
+  CustomerItem,
+  Toast,
+  FormValidation,
+} from '@components';
 
 // Import constant
-import { BUTTON_VARIANTS, BASE_URL, PATH, MESSAGES } from "@constants";
+import { BUTTON_VARIANTS, BASE_URL, PATH, MESSAGES } from '@constants';
 
 // Import list data for Expand component
-import { SORT_TITLES } from "@data";
+import { SORT_TITLES } from '@data';
 
 // Import layout
-import { ProfileInfo } from "@layouts";
+import { ProfileInfo } from '@layouts';
 
 const Analytics = () => {
   // State variables
   const [selectedCustomer, setSelectedCustomer] = useState(null);
   const [isShowProfileInfo, setIsShowProfileInfo] = useState(false);
   const [isShowContextMenu, setIsShowContextMenu] = useState(false);
+  const [isShowForm, setIsShowForm] = useState(false);
+
+  //Handle disable form
+  const handleShowForm = () => {
```

Comment:
check performce for render fucntion

Reply:
![image](https://github.com/datdragon777/React-Training/assets/74806826/9d4fffd2-d5f5-41b4-ab09-3e1b6c3e658c)